samples/python/agents/github-agent/openai_agent_executor.py

```diff
@@ -0,0 +1,241 @@
+import json
+import logging
+from typing import Dict, Any
+
+from openai import AsyncOpenAI
+
+from a2a.server.agent_execution import AgentExecutor
+from a2a.server.agent_execution.context import RequestContext
+from a2a.server.events.event_queue import EventQueue
+from a2a.server.tasks import TaskUpdater
+from a2a.types import (
+    AgentCard,
+    TaskState,
+    TextPart,
+    UnsupportedOperationError,
+)
+from a2a.utils.errors import ServerError
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+class OpenAIAgentExecutor(AgentExecutor):
+    """An AgentExecutor that runs an OpenAI-based Agent."""
+
+    def __init__(self, card: AgentCard, tools: Dict[str, Any], api_key: str):
+        self._card = card
+        self.tools = tools
+        self.client = AsyncOpenAI(
+            api_key=api_key,
+            base_url="https://openrouter.ai/api/v1",
+            default_headers={
+                "HTTP-Referer": "http://localhost:10007",
+                "X-Title": "GitHub Agent"
+            }
+        )
+        self.model = "anthropic/claude-3.5-sonnet"
+        self.system_prompt = """You are a GitHub agent that can help users query information about GitHub repositories and recent project updates.
```

Comment:
This looks like a duplicate of what's in samples/python/agents/github-agent/openai_agent.py . Should you have imported that instead?